controllers/reconcilers/prometheus_installation/prometheus_installation_reconciler.go

```diff
@@ -56,6 +54,25 @@ func (r *Reconciler) Cleanup(ctx context.Context, cr *v1.Observability) (v1.Obse
 		return v1.ResultFailed, err
 	}
 
+	// We have to remove the prometheus operator deployment manually
```

Comment:
This is in the cleanup function, is that intentional, sorry if I misunderstand the change but I understood this was on upgrade issue?